cylc/flow/scripts/reinstall.py

```diff
@@ -0,0 +1,171 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc reinstall [OPTIONS] ARGS
+
+Reinstall a previously installed workflow.
+
+Examples:
+
+ Having previously installed:
+  $ cylc install myflow
+ To reinstall this workflow run:
+  $ cylc reinstall myflow/run1
+
+ Having previously installed:
+ $ cylc install myflow --no-run-name
+ To reinstall this workflow run:
+  $ cylc reinstall myflow
+
+ To reinstall a workflow from within the cylc-run directory of a previously
+ installed workflow:
+ $ cylc reinstall
+
+"""
+
+
+from pathlib import Path
+import pkg_resources
+
+from cylc.flow.exceptions import PluginError, WorkflowFilesError
+from cylc.flow.option_parsers import CylcOptionParser as COP
+from cylc.flow.pathutil import get_workflow_run_dir
+from cylc.flow.platforms import get_platform
+from cylc.flow.suite_files import (
+    get_workflow_source_dir,
+    reinstall_workflow,
+    SuiteFiles)
+from cylc.flow.terminal import cli_function
+
+
+def get_option_parser():
+    parser = COP(
+        __doc__, comms=True, prep=True,
+        argdoc=[("[NAMED_RUN]", "Named run. e.g. my-flow/run1")
+                ])
+
+    # If cylc-rose plugin is available ad the --option/-O config
+    try:
+        __import__('cylc.rose')
+        parser.add_option(
+            "--opt-conf-key", "-O",
+            help=(
+                "Use optional Rose Config Setting"
+                "(If Cylc-Rose is installed)"
+            ),
+            action="append",
+            default=[],
+            dest="opt_conf_keys"
+        )
+        parser.add_option(
+            "--define", '-D',
+            help=(
+                "Each of these overrides the `[SECTION]KEY` setting in a "
+                "`rose-suite.conf` file. "
+                "Can be used to disable a setting using the syntax "
+                "`--define=[SECTION]!KEY` or even `--define=[!SECTION]`."
+            ),
+            action="append",
+            default=[],
+            dest="defines"
+        )
+        parser.add_option(
+            "--define-suite", "--define-flow", '-S',
+            help=(
+                "As `--define`, but with an implicit `[SECTION]` for "
+                "workflow variables."
+            ),
+            action="append",
+            default=[],
+            dest="define_suites"
+        )
+    except ImportError:
+        pass
+
+    return parser
+
+
+@cli_function(get_option_parser)
+def main(parser, opts, named_run=None):
+    if not named_run:
+        source = get_workflow_source_dir(Path.cwd())
```

Comment:
Codecov states that this line (106) is untested. Running `tests/f/cylc-reinstall/00-simple.t` with debug at this line..
![image](https://user-images.githubusercontent.com/37735232/107651454-6f59db80-6c77-11eb-8953-db2faec438ff.png)